### DISCUSSION TITLE
Complex data table example

### DISCUSSION THREAD

**Discussion by lukuser** [Budibase/budibase#9062] [Help] [answered]
Hello, I'm looking for an example or an idea on how I can create a table-like component where 

- Fields can be edited individually.
- Some fields would need to be select lists like the relationship pickers.
- Table would need to be scrollable horizontally, as there'll be around 20 fields there.

I tried with the standard table component, but it seems I can't have relationship pickers there. Also can I edit individual fields on a table? 
I tried adding form components to a table (using it as a container), but then I don't know how to have headers that would scroll together with columns.

**Comment by melohagan** [comment, marked as answer]
Hey @lukuser 

There's nothing like that at the moment, however the [Table block](https://docs.budibase.com/docs/table-block) includes settings for adding Side panels that make performing CRUD operations on your data possible within a single screen.

**Comment by lukuser** [comment]
Hello @melohagan, thank you for the response. Can I use the same panel for updating a row (from the row click) and for adding a new row (from a button)?

**Reply by melohagan** [reply]
There's two settings. 

On row click (open side panel on click for editing a row). 
Show button (open side panel on click of the button for creating a new row).

More here! https://docs.budibase.com/docs/table-block#on-row-click
